src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -142,24 +142,32 @@ SendCommandToWorkersWithMetadataViaSuperUser(const char *command)
 List *
 TargetWorkerSetNodeList(TargetWorkerSet targetWorkerSet, LOCKMODE lockMode)
 {
-	List *workerNodeList = NIL;
-	if (targetWorkerSet == ALL_SHARD_NODES || targetWorkerSet == METADATA_NODES)
-	{
-		workerNodeList = ActivePrimaryNodeList(lockMode);
-	}
-	else
-	{
-		workerNodeList = ActivePrimaryNonCoordinatorNodeList(lockMode);
-	}
+	List *workerNodeList = ActivePrimaryNodeList(lockMode);
 	List *result = NIL;
+	int localGroupId = GetLocalGroupId();
 
 	WorkerNode *workerNode = NULL;
 	foreach_ptr(workerNode, workerNodeList)
 	{
-		if ((targetWorkerSet == NON_COORDINATOR_METADATA_NODES || targetWorkerSet ==
-			 METADATA_NODES) &&
+		if ((targetWorkerSet == OTHER_METADATA_NODES ||
```

Comment:
`AcquireDistributedLockOnRelations_Internal` already handles the local node nicely. 

We have to `sort` the nodes to prevent additional deadlock risk, each node simply starts locking from the first worker node in the list, and the local node fits into the order.

In this patch, we let the local `LOCK` command (or equivalent INDEX command that acquires the lock) to happen after `Preprocess`. That'd mean that different nodes would acquire locks in different nodes, which creates the additional distributed deadlock risk.

In that sense, I think `OTHER_METADATA_NODES` seems like a slightly dangerous concept. The execution piece should be able to handle the order of nodes involved.

I think in this case, we should use `METADATA_NODES`, which already covers all the nodes including the coordinator.

Makes sense?

Reply:
We had a chat offline and discussed why using `METADATA_NODES` here would send the commands to more nodes than required.

Instead we discussed sorting the worker node list in such a way that coordinator is always in the beginning of the list. See #7016. I think this is an orthogonal change that would be helpful for LOCK, TRUNCATE, CREATE/DROP/REINDEX index and other future DDL commands that can be run from workers.